crates/typst-syntax/src/ast.rs

```diff
@@ -1990,6 +1994,24 @@ impl<'a> ModuleImport<'a> {
     }
 }
 
+node! {
+    /// A renamed module import (without items): `import "file.typ" as utils`
+    RenamedModuleImport
+}
+
+impl<'a> RenamedModuleImport<'a> {
+    /// The module or path from which the items should be imported under the
+    /// desired name.
+    pub fn source(self) -> Expr<'a> {
+        self.0.cast_first_match().unwrap_or_default()
+    }
+
+    /// The name to save the module's items under.
+    pub fn new_name(self) -> Ident<'a> {
```

Comment:
this would be moved to `ModuleImport` and return `Option<Ident<'a>>`. you can filter skip_while you don't find a `SyntaxKind::As` child and then find the first match. see `ShowRule` for a similar thing.

Reply:
Gotcha. I only kept it as a separate node because I didn't know how I'd detect a rename here. I'll try your suggestion.

Reply:
Ok, it seems to have worked, thanks! :D